### ISSUE TITLE
bare-string whitelist not working for email

### ISSUE THREAD

**Issue by thagul** [ember-template-lint/ember-template-lint#90]
I'm using ember-i18n but in a template I have an email like `info@domain.com` that I don't translate:

```
...
<div>info@domain.com</div>
...
```

If I use following .template-lintrc.js everything is fine, email is ignored and template linting passes

```
/* jshint node:true */
'use strict';

module.exports = {
  extends: 'recommended',
  rules: {
    'bare-strings': ['info@domain.com']
  }
};
```

Using this configuration, however, I've to whitelist also all chars specified in the default configuration of the `bare-string` rule (`'(', ')', ',', '.', '&', '+', '-', '=', '*', '/', '#', '%', '!', '?', ':', '[', ']', '{', '}'`) but elementAttributes and globalAttributes aren't used any more.

I've tried then with the following config

```
/* jshint node:true */
'use strict';

module.exports = {
  extends: 'recommended',
   rules: {
     'bare-strings': {
       whitelist: ['(', ')', ',', '.', '&', '+', '-', '=', '*', '/', '#', '%', '!', '?', ':', '[', ']', '{', '}', 'info@domain.com'],
       globalAttributes: ['title'],
       elementAttributes: { img: ['alt'], input: ['placeholder'] }
     }      
  }
};
```

but the rule reports `info@domain.com` as not translated text. Also, I've a placeholder in a input textfield with bare string that is not reported.
In the second configuration, if I whitelist `info@domain` and `com` the rule works as expected (email is not reported)

There's something wrong with my configuration or a bug in the bare-string rule?
Thanks

**Comment by rwjblue** [comment]
On initial read it seems like a bug, but I'll try to poke at it a bit deeper to confirm.

**Comment by SharavanaPrasad** [comment]
is there a support for white listing a sentence itself ?
more like this => 

```
 module.exports = {
  extends: 'recommended',
   rules: {
     'bare-strings': {
       whitelist: ['know more'],
       globalAttributes: ['title'],
       elementAttributes: { img: ['alt'], input: ['placeholder'] }
     }      
  }
};
```

 
I tried this way, but still the linter throws an error that 'know more' is not a translated text.. As far as I digged into the code, I believe this should work, but have no idea why it's not working me.

// rules/lint-bare-strings.js

```
  LogStaticStrings.prototype._getBareString = function(_string) {
    var whitelist = this.config.whitelist;
    var string = _string;

    if (whitelist) {
      for (var i = 0; i < whitelist.length; i++) {
        var entry = whitelist[i];

        while (string.indexOf(entry) > -1) {
          string = string.replace(entry, '');
        }
      }
    }

    return string.trim() !== '' ? _string : null;
  };
```

**Comment by initram** [comment]
I am pretty sure this is no longer an issue. The original problem was that we did not order the whitelist by length, so '.' would be replaced before 'info@domain.com'. This is no longer the case, as it will now always try to replace the longest whitelist item first.

**Comment by dcyriller** [comment]
Indeed, thanks @initram!
